cmd/gateway-handlers.go

```diff
@@ -644,15 +644,6 @@ func (api gatewayAPIHandlers) PutBucketHandler(w http.ResponseWriter, r *http.Re
 		return
 	}
 
-	// validating region here, because isValidLocationConstraint
-	// reads body which has been read already. So only validating
-	// region here.
-	serverRegion := serverConfig.GetRegion()
-	if serverRegion != location {
```

Comment:
> Its a behavior problem @krishnasrinivas if user says that this is my region. We are ought to honor it IMHO. For the most part they won't set it. But we willfully ignoring what was set by user is not right.

I see what you mean, if user needs to create a bucket in specific region he should run the gateway in empty reason.

we can use isValidLocation from here:
https://github.com/minio/minio/pull/4329/commits/b58048f1e6ebe980160de40feb819c6a77a931c1